modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscriptionManager.java

```diff
@@ -338,10 +340,14 @@ private void removeLocalSubscriptionAndNotify(AndesSubscription subscription) th
         String messageRouter = destinationType.getAndesMessageRouter();
 
         Query<AndesSubscription> subscriptionQuery = and
-                (equal(AndesSubscription.DURABILITY, isDurable), equal(AndesSubscription.NODE_ID, connectedNode),
-                 equal(AndesSubscription.PROTOCOL, protocolType), equal(AndesSubscription.ROUTER_NAME, messageRouter),
+                (equal(AndesSubscription.DURABILITY, isDurable), equal(AndesSubscription.PROTOCOL, protocolType),
+                 equal(AndesSubscription.ROUTER_NAME, messageRouter),
                  equal(AndesSubscription.ROUTING_KEY, bindingKeyPattern.toLowerCase()));
 
+        if(!SELECT_ALL.equals(connectedNode)){
```

Comment:
If we renamed the constant with something like 'SELECT_ALL_NODES', it would've been more meaningful.